zanata-war/src/main/java/org/zanata/rest/editor/service/resource/SuggestionsResource.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2015, Red Hat, Inc. and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.zanata.rest.editor.service.resource;
+
+import org.zanata.rest.editor.MediaTypes;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+/**
+ * Endpoint to search for suggestions from translation memory and other sources.
+ */
+@Produces({ MediaType.APPLICATION_JSON })
+@Consumes({ MediaType.APPLICATION_JSON })
+public interface SuggestionsResource {
+
+    public static final String SERVICE_PATH = "/suggestions/{sourceLocale}/{transLocale}";
```

Comment:
Yep I like this more: /suggestions?from=en-US&to=fr 

----- Original Message -----

> From: "David Mason" <notifications@github.com>
> To: "zanata/zanata-server" <zanata-server@noreply.github.com>
> Cc: "Patrick Huang" <pahuang@redhat.com>
> Sent: Tuesday, May 19, 2015 3:10:38 PM
> Subject: Re: [zanata-server] REST endpoint for editor translation memory
> (#816)

> In
> zanata-war/src/main/java/org/zanata/rest/editor/service/resource/SuggestionsResource.java
> :
> > +
> > +import org.zanata.rest.editor.MediaTypes;
> > +
> > +import javax.ws.rs.*;
> > +import javax.ws.rs.core.MediaType;
> > +import javax.ws.rs.core.Response;
> > +import java.util.List;
> > +
> > +/**
> > + * Endpoint to search for suggestions from translation memory and other
> > sources.
> > + */
> > +@Produces({ MediaType.APPLICATION_JSON })
> > +@Consumes({ MediaType.APPLICATION_JSON })
> > +public interface SuggestionsResource {
> > +
> > +    public static final String SERVICE_PATH =
> > "/suggestions/{sourceLocale}/{transLocale}";

> My thinking when I chose to make them path parameters was that required
> parameters make more sense to have in the path, and in this case it only
> makes sense to ask for suggestions to translate something from one locale to
> another locale.

> I think the main reason I am attached to the path parameters is that I want
> the URL to look nice and clean, and it looks messier with
> ?sourceLocale=en-US&transLocale=fr than with /en-US/fr . Maybe I can find
> some smaller parameter names to make something that satisfies that desire.

> @huangp what do you think of /suggestions?from=en-US&to=fr ?

> —
> Reply to this email directly or view it on GitHub .